ci/run_tests.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# Workaround for pytest-xdist flaky collection order
+# https://github.com/pytest-dev/pytest/issues/920
+# https://github.com/pytest-dev/pytest/issues/1075
+export PYTHONHASHSEED=$(python -c 'import random; print(random.randint(1, 4294967295))')
+
+if [ -n "$LOCALE_OVERRIDE" ]; then
+    export LC_ALL="$LOCALE_OVERRIDE"
+    export LANG="$LOCALE_OVERRIDE"
+    PANDAS_LOCALE=`python -c 'import pandas; pandas.get_option("display.encoding")'`
+    if [[ "$LOCALE_OVERIDE" != "$PANDAS_LOCALE" ]]; then
+        echo "pandas could not detect the locale. System locale: $LOCALE_OVERRIDE, pandas detected: $PANDAS_LOCALE"
+        # TODO Not really aborting the tests until https://github.com/pandas-dev/pandas/issues/23923 is fixed
+        # exit 1
+    fi
+fi
+if [[ "not network" == *"$PATTERN"* ]]; then
+    export http_proxy=http://1.2.3.4 https_proxy=http://1.2.3.4;
+fi
+
+
+if [ -n "$PATTERN" ]; then
+    PATTERN=" and $PATTERN"
+fi
+
+for TYPE in single multiple
+do
+    if [ "$COVERAGE" ]; then
+        COVERAGE_FNAME="/tmp/coc-$TYPE.xml"
+        COVERAGE="-s --cov=pandas --cov-report=xml:$COVERAGE_FNAME"
+    fi
+
+    TYPE_PATTERN=$TYPE
+    NUM_JOBS=1
+    if [[ "$TYPE_PATTERN" == "multiple" ]]; then
+        TYPE_PATTERN="not single"
+        NUM_JOBS=2
+    fi
+
+    pytest -m "$TYPE_PATTERN$PATTERN" -n $NUM_JOBS -s --strict --durations=10 --junitxml=test-data-$TYPE.xml $TEST_ARGS $COVERAGE pandas
```

Comment:
you have TEST_ARGS here, though you removed it elsewhere

Reply:
I wanted to get rid of it, but it's needed in the numpydev build for the `-W error`

Reply:
any other way to do that. this makes it pretty clunky.

Reply:
I want to take a look at pytest concurrency in a different PR.

I think it should be a simpler way to avoid running the `simple` tests in parallel, without having to call pytest twice. I think that would be cleaner, and we can simplify all this even more.

Also, I want to see how many processes we have, and not call always for 2 (which I assume it's the number of cores in Travis), but instead use all the available (I think that should be as easy as `-n auto`, but I want to take a look at how many cores are detected in azure, and how many makes sense to use).

Reply:
the single tests must remain though
we got failures because these are using single system resources